src/ORM/Query.php

```diff
@@ -1129,7 +1136,21 @@ protected function _execute(): iterable
         $results = $this->execute()->fetchAll(StatementInterface::FETCH_TYPE_ASSOC);
         $results = $this->getEagerLoader()->loadExternal($this, $results);
 
-        return new ResultSet($this, $results);
+        return $this->resultSetFactory()->createResultSet($this, $results);
+    }
+
+    /**
+     * Get resultset factory.
+     *
+     * @return \Cake\ORM\ResultSetFactory
+     */
+    protected function resultSetFactory(): ResultSetFactory
+    {
+        if (isset($this->resultSetFactory)) {
```

Comment:
should this be !== null instead?

Reply:
No, that will generate a "cannot access uninitialized typed property" error.

Reply:
Oh, this new php8 stuff ;) gotcha

Reply:
I had mistakenly made the property nullable, removed it now.

Reply:
In general there are various properties property throughout our codebase which are never set to `null` after being initialized. They all need to be non-nullable.